src/yunohost/user.py

```diff
@@ -462,6 +469,276 @@ def user_info(auth, username):
     else:
         raise YunohostError('user_info_failed')
 
+#
+# Group subcategory
+#
+#
+def user_group_list(auth, fields=None):
+    """
+    List users
+
+    Keyword argument:
+        filter -- LDAP filter used to search
+        offset -- Starting number for user fetching
+        limit -- Maximum number of user fetched
+        fields -- fields to fetch
+
+    """
+    group_attr = {
+        'cn' : 'groupname',
+        'member' : 'members',
+        'permission' : 'permission'
+    }
+    attrs = ['cn']
+    groups = {}
+
+    if fields:
+        keys = group_attr.keys()
+        for attr in fields:
+            if attr in keys:
+                attrs.append(attr)
+            else:
+                raise YunohostError('field_invalid', attr)
+    else:
+        attrs = ['cn', 'member']
+
+    result = auth.search('ou=groups,dc=yunohost,dc=org',
+                         '(objectclass=groupOfNamesYnh)',
+                         attrs)
+
+    for group in result:
+        # The group "admins" should be hidden for the user
+        if group_attr['cn'] == "admins":
+            continue
+        entry = {}
+        for attr, values in group.items():
+            if values:
+                if attr == "member":
+                    entry[group_attr[attr]] = []
+                    for v in values:
+                            entry[group_attr[attr]].append(v.split("=")[1].split(",")[0])
+                elif attr == "permission":
+                    entry[group_attr[attr]] = {}
+                    for v in values:
+                        permission = v.split("=")[1].split(",")[0].split(".")[1]
+                        pType = v.split("=")[1].split(",")[0].split(".")[0]
+                        if permission in entry[group_attr[attr]]:
+                            entry[group_attr[attr]][permission].append(pType)
+                        else:
+                            entry[group_attr[attr]][permission] = [pType]
+                else:
+                    entry[group_attr[attr]] = values[0]
+
+        groupname = entry[group_attr['cn']]
+        groups[groupname] = entry
+    return {'groups' : groups}
+
+
+@is_unit_operation([('groupname', 'user')])
```

Comment:
Uh it's not clear to me how this `is_unit_operation` works but eh, what `'user'` are we talking about ? Maybe something missing here, idk... #drunk